AntennaTracker/tracking.cpp

```diff
@@ -143,14 +146,63 @@ void Tracker::tracking_update_position(const mavlink_global_position_int_t &msg)
     vehicle.location.alt = msg.alt/10;
     vehicle.relative_alt = msg.relative_alt/10;
     vehicle.vel = Vector3f(msg.vx/100.0f, msg.vy/100.0f, msg.vz/100.0f);
-    vehicle.last_update_us = AP_HAL::micros();
+
+    const uint32_t now = AP_HAL::micros();
+    vehicle.last_update_us = now;
+    vehicle.mavlink_last_update_us = now;
+
     vehicle.last_update_ms = AP_HAL::millis();
+
     // log vehicle as GPS2
     if (should_log(MASK_LOG_GPS)) {
         Log_Write_Vehicle_Pos(vehicle.location.lat, vehicle.location.lng, vehicle.location.alt, vehicle.vel);
     }
 }
 
+// update from ADS-B
+void Tracker::adsb_tracking_update_position()
+{
+    // check for valid target
+    const uint32_t adsb_target = adsb.get_special_ICAO_target();
+    if (adsb_target == 0) {
+        return;
+    }
+
+    // try and read target
+    AP_ADSB::adsb_vehicle_t adsb_vehicle;
+    if (!adsb.get_vehicle_by_ICAO(adsb_target, adsb_vehicle)) {
+        return;
+    }
+
+    Location temp_location;
```

Comment:
Doing this will save you a few lines:
`Location temp_location = adsb.get_location(adsb_vehicle);`